daemons/pacemakerd/pcmkd_subdaemons.c

```diff
@@ -482,21 +480,25 @@ start_child(pcmk_child_t * child)
         if(gid) {
             // Drop root group access if not needed
             if (!need_root_group && (setgid(gid) < 0)) {
-                crm_warn("Could not set group to %d: %s", gid, strerror(errno));
+                crm_warn("Could not set subdaemon %s group to %lu: %s",
+                         name, (unsigned long) gid, strerror(errno));
             }
 
             /* Initialize supplementary groups to only those always granted to
              * the user, plus haclient (so we can access IPC).
              */
             if (initgroups(child->uid, gid) < 0) {
-                crm_err("Cannot initialize groups for %s: %s (%d)",
-                        child->uid, pcmk_rc_str(errno), errno);
+                crm_err("Cannot initialize system groups for subdaemon %s: %s "
+                        QB_XS " errno=%d",
+                        name, pcmk_rc_str(errno), errno);
             }
         }
 
         if (uid && setuid(uid) < 0) {
-            crm_warn("Could not set user to %s (id %d): %s",
-                     child->uid, uid, strerror(errno));
+            crm_warn("Could not set subdaemon %s user to %s: %s "
+                     QB_XS " uid=%lu errno=%d",
+                     name, child->uid, (unsigned long) uid, strerror(errno),
```

Comment:
still need to swap `(unsigned long) uid` and `strerror(errno)`

Reply:
whoops, fixed